src/Features/CSharp/CodeFixes/Async/CSharpAddAsyncCodeFixProvider.cs

```diff
@@ -62,53 +61,40 @@ protected override string GetDescription(Diagnostic diagnostic, SyntaxNode node,
 
         private static SyntaxNode GetContainingMember(SyntaxNode oldNode)
         {
-            var parenthesizedLambda = oldNode
-                .Ancestors()
-                .FirstOrDefault(n =>
-                    n.IsKind(SyntaxKind.ParenthesizedLambdaExpression));
-
-            if (parenthesizedLambda != null)
-            {
-                return parenthesizedLambda;
-            }
-
-            var simpleLambda = oldNode
-                .Ancestors()
-                .FirstOrDefault(n =>
-                    n.IsKind(SyntaxKind.SimpleLambdaExpression));
-
-            if (simpleLambda != null)
+            foreach (var node in oldNode.Ancestors())
             {
-                return simpleLambda;
+                switch (node.Kind())
+                {
+                    case SyntaxKind.ParenthesizedLambdaExpression:
+                    case SyntaxKind.SimpleLambdaExpression:
+                    case SyntaxKind.AnonymousMethodExpression:
+                        if ((node as AnonymousFunctionExpressionSyntax)?.AsyncKeyword != SyntaxFactory.Token(SyntaxKind.AsyncKeyword))
```

Comment:
Why not just check AsyncKeyword.Kind rather than constructing a new token?

```C#
if ((node as AnonymousFunctionExpressionSyntax)?.AsyncKeyword.Kind != SyntaxKind.AsyncKeyword)
```